cmd/server.go

```diff
@@ -141,12 +143,14 @@ func createHTTPServer(gormDB *gorm.DB) *echo.Echo {
 	if api.UpstreamConf.IsPartiallyFilled() {
 		logger.Warnf("Please ensure that all the required flags for upstream is supplied.")
 	}
-	upstreamGroup := e.Group("/upstream")
+	upstreamGroup := e.Group("/upstream", rbac.Authorization(rbac.ObjectAgentPush, rbac.ActionWrite))
 	upstreamGroup.POST("/push", upstream.PushUpstream)
 	upstreamGroup.GET("/pull/:agent_name", upstream.Pull)
 	upstreamGroup.GET("/canary/pull/:agent_name", canary.Pull)
 	upstreamGroup.GET("/status/:agent_name", upstream.Status)
 
+	e.POST("/agent/generate", agent.GenerateAgent)
```

Comment:
@moshloop Should this be added to exclusion in the kratos middleware?
In here https://github.com/flanksource/incident-commander/blob/main/auth/middleware.go#L48

Reply:
No - generating an agent should be restricted to the admin role